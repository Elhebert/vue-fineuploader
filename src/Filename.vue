<template>
  <renderless-file-name
    :id="id"
    :uploader="uploader"
   >
    <span slot-scope="{ filename }">{{ filename }}</span>
  </renderless-file-name>
</template>

<script>
import RenderlessFileName from './renderless/Filename'

export default {
  components: {
    RenderlessFileName,
  },

  props: {
    id: {
      type: Number,
      required: true,
    },
    uploader: {
      type: Object,
      required: true,
    },
  },
}
</script>
