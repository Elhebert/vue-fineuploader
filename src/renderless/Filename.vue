<script>
export default {
  props: {
    id: {
      type: Number,
      required: true,
    },
    uploader: {
      type: Object,
      required: true,
    },
  },

  data: () => ({
    filename: '',
  }),

  created() {
    this.filename = this.uploader.methods.getName(this.id)
    this.interceptSetName()
  },

  methods: {
    interceptSetName() {
      const oldSetName = this.uploader.methods.setName

      this.uploader.methods.setName = (id, newName) => {
        oldSetName.call(this.uploader.methods, id, newName)

        if (this.isCurrentFile(id)) {
          this.filename = newName
        }
      }
    },

    isCurrentFile(id) {
      return this.id === id
    },
  },

  render() {
    return this.$scopedSlots.default({
      filename: this.filename,
    })
  },
}
</script>
