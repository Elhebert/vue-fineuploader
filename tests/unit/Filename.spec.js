import { mount } from '@vue/test-utils'
import Filename from '@/Filename.vue'
import FineUploader from 'fine-uploader-wrappers'

describe('Filename.vue', () => {
  let uploader
  let wrapper

  const sampleBlob = new Blob(['hi!'], { type: 'text/plain' })
  const sampleBlobWrapper = { blob: sampleBlob, name: 'test' }

  beforeEach(done => {
    uploader = new FineUploader({ options: { autoUpload: false } })
    uploader.on('submitted', done)
    uploader.methods.addFiles(sampleBlobWrapper)

    wrapper = mount(Filename, {
      propsData: { id: 0, uploader },
    })
  })

  it('renders initial filename', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.text()).toBe('test')
  })

  it('updates filename on setName', () => {
    uploader.methods.setName(0, 'new-name')

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.text()).toBe('new-name')
  })
})
