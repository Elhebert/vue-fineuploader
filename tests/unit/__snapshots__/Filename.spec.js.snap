// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filename.vue renders initial filename 1`] = `<span>test</span>`;

exports[`Filename.vue updates filename on setName 1`] = `<span>new-name</span>`;
